osu.Game/Overlays/DirectOverlay.cs

```diff
@@ -286,6 +286,15 @@ private void updateSearch()
 
         private int distinctCount(List<string> list) => list.Distinct().ToArray().Length;
 
+        public void ShowTextSearch(string searchText)
+        {
+            currentQuery.Value = searchText;
+
+            // Hide first to put the direct overlay into the foreground if its behind another overlay.
+            Hide();
+            Show();
+        }
```

Comment:
Is there a more elegant way to do this? It feels wrong but functions correctly.

Reply:
I'm going to fix this one in a separate PR. Requires framework side changes too.

Reply:
Please check #4989 works as you expect.

Reply:
Looks good. Thanks for taking care of it.